[CI_Block]num_entities of partition is not the same with inserted quantities under multiple replicas

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
num_entities of partition is not the same with inserted quantity under multiple replicas when data is relatively big

**Steps/Code to reproduce behavior**
1 deploy multiple replicas enviroment:
   proxy: 1, querynode:2, indexnode:2, datanode: 2

2 run test case: tests20/python_client/testcases/test_partition.py::TestPartitionOperations::test_partition_insert_maximum_size_data

Result:
assert partition_w.num_entities == max_size
E       assert 77090 == 100000
E         +77090
E         -100000

Logs:
[log.tar.zip](https://github.com/milvus-io/milvus/files/6799201/log.tar.zip)

**Expected behavior**
num_entities of partition equals to inserted quantities

**Method of installation**
- [ ] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Software conditions (OS, CPU, GPU, Memory)


- Milvus version (master or released version)
milvus-master (a8e5fd2)
Name: pymilvus
Version: 2.0.0rc2.dev11
Name: pymilvus-orm
Version: 2.0.0rc2.dev29

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

in CI tests(no replicas), if insert 100,000 entities, it repros as well:
  ```
# insert data to partition
        max_size = 100000  # TODO: clarify the max size of data
        ins_res, succ = partition_w.insert(cf.gen_default_dataframe_data(max_size))
        assert len(ins_res.primary_keys) == max_size
        assert succ
        # self._connect().flush([collection_w.name])
>       assert partition_w.num_entities == max_size
E       assert 98618 == 100000
```
It looks loke flush is not completed when calling .num_entities. Does flush run as sync or async by default?

it seems to be getting worse and worse, this time it repros by 3000 entities in CI tests(only in standalone pipeline):

stats = connect.get_collection_stats(collection_list[i])=
[2021-07-16T08:14:07.991Z] >           assert stats[row_count] == default_nb
[2021-07-16T08:14:07.991Z] E           assert 2064 == 3000
[2021-07-16T08:14:07.991Z] E             +2064
[2021-07-16T08:14:07.991Z] E             -3000
[2021-07-16T08:14:07.991Z] 
[2021-07-16T08:14:07.991Z] collection/test_collection_stats.py:370: AssertionError
[2021-07-16T08:14:07.991Z] =========================== short test summary info ============================
[2021-07-16T08:14:07.991Z] FAILED collection/test_collection_stats.py::TestGetCollectionStats::test_collection_count_multi_collections

```
[2021-07-23T02:39:33.993Z] self = <test_collection_count.TestCollectionMultiCollections object at 0x7faf593a10b8>

[2021-07-23T02:39:33.993Z] connect = <pymilvus.client.stub.Milvus object at 0x7faf5abca630>

[2021-07-23T02:39:33.993Z] 

[2021-07-23T02:39:33.993Z]     @pytest.mark.level(2)

[2021-07-23T02:39:33.993Z]     @pytest.mark.tags(CaseLabel.tags_smoke)

[2021-07-23T02:39:33.993Z]     def test_collection_count_multi_collections_mix(self, connect):

[2021-07-23T02:39:33.993Z]         '''

[2021-07-23T02:39:33.993Z]         target: test collection rows_count is correct or not with multiple collections of JACCARD

[2021-07-23T02:39:33.993Z]         method: create collection and add entities in it,

[2021-07-23T02:39:33.993Z]             assert the value returned by count_entities method is equal to length of entities

[2021-07-23T02:39:33.993Z]         expected: the count is equal to the length of entities

[2021-07-23T02:39:33.993Z]         '''

[2021-07-23T02:39:33.993Z]         collection_list = []

[2021-07-23T02:39:33.993Z]         collection_num = 20

[2021-07-23T02:39:33.993Z]         for i in range(0, int(collection_num / 2)):

[2021-07-23T02:39:33.993Z]             collection_name = gen_unique_str(uid)

[2021-07-23T02:39:33.993Z]             collection_list.append(collection_name)

[2021-07-23T02:39:33.993Z]             connect.create_collection(collection_name, default_fields)

[2021-07-23T02:39:33.993Z]             connect.insert(collection_name, default_entities)

[2021-07-23T02:39:33.993Z]         for i in range(int(collection_num / 2), collection_num):

[2021-07-23T02:39:33.993Z]             collection_name = gen_unique_str(uid)

[2021-07-23T02:39:33.993Z]             collection_list.append(collection_name)

[2021-07-23T02:39:33.993Z]             connect.create_collection(collection_name, default_binary_fields)

[2021-07-23T02:39:33.993Z]             res = connect.insert(collection_name, default_binary_entities)

[2021-07-23T02:39:33.993Z]         connect.flush(collection_list)

[2021-07-23T02:39:33.993Z]         for i in range(collection_num):

[2021-07-23T02:39:33.993Z]             stats = connect.get_collection_stats(collection_list[i])

[2021-07-23T02:39:33.993Z] >           assert stats[row_count] == default_nb

[2021-07-23T02:39:33.993Z] E           assert 2064 == 3000

[2021-07-23T02:39:33.993Z] E             +2064

[2021-07-23T02:39:33.993Z] E             -3000

[2021-07-23T02:39:33.993Z] 

[2021-07-23T02:39:33.993Z] collection/test_collection_count.py:499: AssertionError

[2021-07-23T02:39:33.993Z] =========================== short test summary info ============================

[2021-07-23T02:39:33.993Z] FAILED collection/test_collection_count.py::TestCollectionMultiCollections::test_collection_count_multi_collections_mix
```

--- a/internal/datacoord/segment_manager.go
+++ b/internal/datacoord/segment_manager.go
@@ -376,7 +376,7 @@ func (s *SegmentManager) GetFlushableSegments(ctx context.Context, channel strin
 	defer s.mu.Unlock()
 	sp, _ := trace.StartSpanFromContext(ctx)
 	defer sp.Finish()
-	if err := s.tryToSealSegment(t); err != nil {
+	if err := s.tryToSealSegment(t, channel); err != nil {
 		return nil, err
 	}
 
@@ -384,6 +384,9 @@ func (s *SegmentManager) GetFlushableSegments(ctx context.Context, channel strin
 	for _, id := range s.segments {
 		info := s.meta.GetSegment(id)
 		if info == nil {
+			continue
+		}
+		if info.InsertChannel != channel {
 			continue
 		}
 		if s.flushPolicy(info, t) {
@@ -403,6 +406,9 @@ func (s *SegmentManager) ExpireAllocations(channel string, ts Timestamp) error {
 		if segment == nil {
 			continue
 		}
+		if segment.InsertChannel != channel {
+			continue
+		}
 		for i := 0; i < len(segment.allocations); i++ {
 			if segment.allocations[i].ExpireTime <= ts {
 				a := segment.allocations[i]
@@ -416,12 +422,15 @@ func (s *SegmentManager) ExpireAllocations(channel string, ts Timestamp) error {
 }
 
 // tryToSealSegment applies segment & channel seal policies
-func (s *SegmentManager) tryToSealSegment(ts Timestamp) error {
+func (s *SegmentManager) tryToSealSegment(ts Timestamp, channel string) error {
 	channelInfo := make(map[string][]*SegmentInfo)
 	for _, id := range s.segments {
 		info := s.meta.GetSegment(id)
 		if info == nil {
 			log.Warn("Failed to get seg info from meta", zap.Int64("id", id))
+			continue
+		}
+		if info.InsertChannel != channel {
 			continue
 		}
 		channelInfo[info.InsertChannel] = append(channelInfo[info.InsertChannel], info)

--- a/internal/datacoord/segment_manager.go
+++ b/internal/datacoord/segment_manager.go
@@ -383,10 +383,7 @@ func (s *SegmentManager) GetFlushableSegments(ctx context.Context, channel strin
 	ret := make([]UniqueID, 0, len(s.segments))
 	for _, id := range s.segments {
 		info := s.meta.GetSegment(id)
-		if info == nil {
-			continue
-		}
-		if info.InsertChannel != channel {
+		if info == nil || info.InsertChannel != channel {
 			continue
 		}
 		if s.flushPolicy(info, t) {
@@ -403,10 +400,7 @@ func (s *SegmentManager) ExpireAllocations(channel string, ts Timestamp) error {
 	defer s.mu.Unlock()
 	for _, id := range s.segments {
 		segment := s.meta.GetSegment(id)
-		if segment == nil {
-			continue
-		}
-		if segment.InsertChannel != channel {
+		if segment == nil || segment.InsertChannel != channel {
 			continue
 		}
 		for i := 0; i < len(segment.allocations); i++ {
@@ -426,11 +420,7 @@ func (s *SegmentManager) tryToSealSegment(ts Timestamp, channel string) error {
 	channelInfo := make(map[string][]*SegmentInfo)
 	for _, id := range s.segments {
 		info := s.meta.GetSegment(id)
-		if info == nil {
-			log.Warn("Failed to get seg info from meta", zap.Int64("id", id))
-			continue
-		}
-		if info.InsertChannel != channel {
+		if info == nil || info.InsertChannel != channel {
 			continue
 		}
 		channelInfo[info.InsertChannel] = append(channelInfo[info.InsertChannel], info)

--- a/internal/datacoord/server.go
+++ b/internal/datacoord/server.go
@@ -335,8 +335,6 @@ func (s *Server) startDataNodeTtLoop(ctx context.Context) {
 
 			ch := ttMsg.ChannelName
 			ts := ttMsg.Timestamp
-			// log.Debug("Receive datanode timetick msg", zap.String("channel", ch),
-			// zap.Any("ts", ts))
 			segments, err := s.segmentManager.GetFlushableSegments(ctx, ch, ts)
 			if err != nil {
 				log.Warn("get flushable segments failed", zap.Error(err))

--- a/internal/proxy/channels_time_ticker.go
+++ b/internal/proxy/channels_time_ticker.go
@@ -96,7 +96,6 @@ func (ticker *channelsTimeTickerImpl) tick() error {
 		log.Warn("Proxy channelsTimeTickerImpl failed to get ts from tso", zap.Error(err))
 		return err
 	}
-	//nowPTime, _ := tsoutil.ParseTS(now)
 
 	ticker.statisticsMtx.Lock()
 	defer ticker.statisticsMtx.Unlock()
@@ -111,36 +110,19 @@ func (ticker *channelsTimeTickerImpl) tick() error {
 
 	for pchan := range ticker.currents {
 		current := ticker.currents[pchan]
-		//currentPTime, _ := tsoutil.ParseTS(current)
 		stat, ok := stats[pchan]
-		//log.Debug("Proxy channelsTimeTickerImpl", zap.Any("pchan", pchan),
-		//	zap.Any("TaskInQueue", ok),
-		//	zap.Any("current", currentPTime),
-		//	zap.Any("now", nowPTime))
+
 		if !ok {
 			ticker.minTsStatistics[pchan] = current
 			ticker.currents[pchan] = now
 		} else {
-			//minPTime, _ := tsoutil.ParseTS(stat.minTs)
-			//maxPTime, _ := tsoutil.ParseTS(stat.maxTs)
-
 			if stat.minTs > current {
-				cur := current
-				if stat.minTs > now {
-					cur = now
-				}
-				ticker.minTsStatistics[pchan] = cur
+				ticker.minTsStatistics[pchan] = stat.minTs - 1
 				next := now + Timestamp(sendTimeTickMsgInterval)
 				if next > stat.maxTs {
 					next = stat.maxTs
 				}
 				ticker.currents[pchan] = next
-				//nextPTime, _ := tsoutil.ParseTS(next)
-				//log.Debug("Proxy channelsTimeTickerImpl",
-				//	zap.Any("pchan", pchan),
-				//	zap.Any("minPTime", minPTime),
-				//	zap.Any("maxPTime", maxPTime),
-				//	zap.Any("nextPTime", nextPTime))
 			}
 		}
 	}
@@ -190,7 +172,6 @@ func (ticker *channelsTimeTickerImpl) close() error {
 
 func (ticker *channelsTimeTickerImpl) addPChan(pchan pChan) error {
 	ticker.statisticsMtx.Lock()
-
 	if _, ok := ticker.minTsStatistics[pchan]; ok {
 		ticker.statisticsMtx.Unlock()
 		return fmt.Errorf("pChan %v already exist in minTsStatistics", pchan)

--- a/internal/proxy/channels_time_ticker_test.go
+++ b/internal/proxy/channels_time_ticker_test.go
@@ -11,7 +11,6 @@
 
 package proxy
 
-/*
 import (
 	"context"
 	"math/rand"
@@ -38,7 +37,8 @@ func newMockTsoAllocator() *mockTsoAllocator {
 }
 
 func newGetStatisticsFunc(pchans []pChan) getPChanStatisticsFuncType {
-	pchanNum := rand.Uint64()%5 + 1
+	total_pchan := len(pchans)
+	pchanNum := rand.Uint64()%(uint64(total_pchan)) + 1
 	pchans2 := make([]pChan, 0, pchanNum)
 	for i := 0; uint64(i) < pchanNum; i++ {
 		pchans2 = append(pchans2, pchans[i])
@@ -223,4 +223,3 @@ func TestChannelsTimeTickerImpl_getMinTsStatistics(t *testing.T) {
 
 	time.Sleep(time.Second)
 }
-*/

--- a/internal/proxy/channels_time_ticker_test.go
+++ b/internal/proxy/channels_time_ticker_test.go
@@ -37,8 +37,8 @@ func newMockTsoAllocator() *mockTsoAllocator {
 }
 
 func newGetStatisticsFunc(pchans []pChan) getPChanStatisticsFuncType {
-	total_pchan := len(pchans)
-	pchanNum := rand.Uint64()%(uint64(total_pchan)) + 1
+	totalPchan := len(pchans)
+	pchanNum := rand.Uint64()%(uint64(totalPchan)) + 1
 	pchans2 := make([]pChan, 0, pchanNum)
 	for i := 0; uint64(i) < pchanNum; i++ {
 		pchans2 = append(pchans2, pchans[i])

--- a/internal/proxy/task.go
+++ b/internal/proxy/task.go
@@ -99,6 +99,7 @@ type dmlTask interface {
 	task
 	getChannels() ([]vChan, error)
 	getPChanStats() (map[pChan]pChanStatistics, error)
+	getChannelsTimerTicker() channelsTimeTicker
 }
 
 type BaseInsertTask = msgstream.InsertMsg
@@ -151,6 +152,10 @@ func (it *InsertTask) SetTs(ts Timestamp) {
 
 func (it *InsertTask) EndTs() Timestamp {
 	return it.EndTimestamp
+}
+
+func (it *InsertTask) getChannelsTimerTicker() channelsTimeTicker {
+	return it.chTicker
 }
 
 func (it *InsertTask) getPChanStats() (map[pChan]pChanStatistics, error) {
@@ -1023,15 +1028,6 @@ func (it *InsertTask) Execute(ctx context.Context) error {
 			it.result.Status.Reason = err.Error()
 			return err
 		}
-	}
-
-	pchans, err := it.chMgr.getChannels(collID)
-	if err != nil {
-		return err
-	}
-	for _, pchan := range pchans {
-		log.Debug("Proxy InsertTask add pchan", zap.Any("pchan", pchan))
-		_ = it.chTicker.addPChan(pchan)
 	}
 
 	// Assign SegmentID

--- a/internal/proxy/task_scheduler.go
+++ b/internal/proxy/task_scheduler.go
@@ -224,7 +224,6 @@ func (queue *DmTaskQueue) Enqueue(t task) error {
 	if err != nil {
 		return err
 	}
-
 	_ = queue.addPChanStats(t)
 
 	return nil
@@ -248,10 +247,10 @@ func (queue *DmTaskQueue) addPChanStats(t task) error {
 	if dmT, ok := t.(dmlTask); ok {
 		stats, err := dmT.getPChanStats()
 		if err != nil {
+			log.Debug("Proxy DmTaskQueue addPChanStats", zap.Any("tID", t.ID()),
+				zap.Any("stats", stats), zap.Error(err))
 			return err
 		}
-		log.Debug("Proxy DmTaskQueue addPChanStats", zap.Any("tID", t.ID()),
-			zap.Any("stats", stats))
 		queue.statsLock.Lock()
 		for cName, stat := range stats {
 			info, ok := queue.pChanStatisticsInfos[cName]
@@ -263,6 +262,7 @@ func (queue *DmTaskQueue) addPChanStats(t task) error {
 					},
 				}
 				queue.pChanStatisticsInfos[cName] = info
+				dmT.getChannelsTimerTicker().addPChan(cName)
 			} else {
 				if info.minTs > stat.minTs {
 					queue.pChanStatisticsInfos[cName].minTs = stat.minTs
@@ -275,7 +275,7 @@ func (queue *DmTaskQueue) addPChanStats(t task) error {
 		}
 		queue.statsLock.Unlock()
 	} else {
-		return fmt.Errorf("Proxy addUnissuedTask reflect to dmlTask failed, tID:%v", t.ID())
+		return fmt.Errorf("proxy addUnissuedTask reflect to dmlTask failed, tID:%v", t.ID())
 	}
 	return nil
 }

